skrifa/src/meta/charmap.rs

```diff
@@ -0,0 +1,374 @@
+                        cmap14 = Some(subtable);
+                    }
+                }
+            }
+            (PlatformId::Windows, ENCODING_MS_SYMBOL) => {
+                // Symbol
+                if let Some(subtable) = UnicodeSubtable::from_cmap_record(cmap, record) {
+                    maybe_choose_subtable(MapKind::Symbol, i, subtable);
+                }
+            }
+            (PlatformId::Windows, ENCODING_MS_ID_UCS_4)
+            | (PlatformId::Unicode, ENCODING_APPLE_ID_UNICODE_32) => {
+                // Unicode full repertoire
+                if let Some(subtable) = UnicodeSubtable::from_cmap_record(cmap, record) {
+                    maybe_choose_subtable(MapKind::UnicodeFull, i, subtable);
+                }
+            }
+            (PlatformId::ISO, _)
+            | (PlatformId::Unicode, _)
+            | (PlatformId::Windows, ENCODING_MS_UNICODE_CS) => {
+                // Unicode BMP only
+                if let Some(subtable) = UnicodeSubtable::from_cmap_record(cmap, record) {
+                    maybe_choose_subtable(MapKind::UnicodeBmp, i, subtable);
+                }
+            }
+            _ => {}
+        }
+    }
+    (maps, map, cmap14)
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use crate::MetadataProvider;
+    use read_fonts::FontRef;
+
+    #[test]
+    fn choose_format_12_over_4() {
+        let font = FontRef::new(read_fonts::test_data::test_fonts::CMAP12_FONT1).unwrap();
+        let charmap = font.charmap();
+        assert!(matches!(
+            charmap.map.unwrap().subtable,
+            UnicodeSubtable::Format12(..)
+        ));
+    }
+
+    #[test]
+    fn choose_format_4() {
+        let font = FontRef::new(read_fonts::test_data::test_fonts::VAZIRMATN_VAR).unwrap();
+        let charmap = font.charmap();
+        assert!(matches!(
+            charmap.map.unwrap().subtable,
+            UnicodeSubtable::Format4(..)
+        ));
+    }
+
+    #[test]
+    fn map_format_4() {
+        let font = FontRef::new(read_fonts::test_data::test_fonts::VAZIRMATN_VAR).unwrap();
+        let charmap = font.charmap();
+        assert_eq!(charmap.map('A'), Some(GlyphId::new(1)));
+        assert_eq!(charmap.map('À'), Some(GlyphId::new(2)));
+        assert_eq!(charmap.map('`'), Some(GlyphId::new(3)));
+        assert_eq!(charmap.map('B'), None);
+    }
+
+    #[test]
+    fn map_format_12() {
+        let font = FontRef::new(read_fonts::test_data::test_fonts::CMAP12_FONT1).unwrap();
+        let charmap = font.charmap();
+        assert_eq!(charmap.map(' '), None);
+        assert_eq!(charmap.map(0x101723_u32), Some(GlyphId::new(23)));
+        assert_eq!(charmap.map(0x101725_u32), Some(GlyphId::new(25)));
+        assert_eq!(charmap.map(0x102523_u32), Some(GlyphId::new(53)));
+        assert_eq!(charmap.map(0x102526_u32), Some(GlyphId::new(56)));
+        assert_eq!(charmap.map(0x102527_u32), Some(GlyphId::new(57)));
+    }
+
+    #[test]
+    fn map_symbol_pua() {
```

Comment:
You're describing above that symbol encoding will get selected over a regular encoding - can we cover that in tests as well?

Reply:
Good catch! I’ll add a non symbol subtable to the test font and an additional test to ensure the correct subtable is selected.